sdk/tables/azure-data-tables/tests/test_custom_entity.py

```diff
@@ -0,0 +1,422 @@
+            u"Birthday": u"2020-01-01T12:59:59.0123456Z",
+        }
+        try:
+            with table_client as tc:
+                tc.create_entity(entity)
+                entity = tc.get_entity(
+                    partition_key=entity[u"PartitionKey"],
+                    row_key=entity[u"RowKey"],
+                    entity_hook=MyEntity,
+                )
+
+                assert isinstance(entity.properties["Birthday"], datetime)
+                assert "milliseconds" in entity.keys()
+                assert entity.properties["milliseconds"] == 0.0123456
+                assert entity.properties["Birthday"] == datetime(2020, 1, 1, 12, 59, 59)
+
+        finally:
+            self._tear_down(
+                tables_storage_account_name, tables_primary_storage_account_key
+            )
+
+    @TablesPreparer()
+    def test_custom_entity_list(
+        self, tables_storage_account_name, tables_primary_storage_account_key
+    ):
+        table_client = self._set_up(
+            tables_storage_account_name, tables_primary_storage_account_key
+        )
+
+        entity = {
+            u"PartitionKey": u"pk",
+            u"RowKey": u"rk",
+            u"Birthday": u"2020-01-01T12:59:59.0123456Z",
+        }
+        entity2 = {
+            u"PartitionKey": u"pk",
+            u"RowKey": u"rk1",
+            u"Birthday": u"2021-02-02T12:59:59.0123456Z",
+        }
+        try:
+            with table_client as tc:
+                tc.create_entity(entity)
+                tc.create_entity(entity2)
+                entities = tc.list_entities(entity_hook=MyEntity)
+
+                length = 0
+                for entity in entities:
+                    assert isinstance(entity.properties["Birthday"], datetime)
+                    assert "milliseconds" in entity.keys()
+                    assert entity.properties["milliseconds"] == 0.0123456
+                    length += 1
+                assert length == 2
+
+        finally:
+            self._tear_down(
+                tables_storage_account_name, tables_primary_storage_account_key
+            )
+
+    @TablesPreparer()
+    def test_custom_entity_query(
+        self, tables_storage_account_name, tables_primary_storage_account_key
+    ):
+        table_client = self._set_up(
+            tables_storage_account_name, tables_primary_storage_account_key
+        )
+
+        entity = {
+            u"PartitionKey": u"pk",
+            u"RowKey": u"rk",
+            u"Birthday": u"2020-01-01T12:59:59.0123456Z",
+            u"Value": 20,
+        }
+        entity2 = {
+            u"PartitionKey": u"pk",
+            u"RowKey": u"rk1",
+            u"Birthday": u"2021-02-02T12:59:59.0123456Z",
+            u"Value": 30,
+        }
+        try:
+            with table_client as tc:
+                tc.create_entity(entity)
+                tc.create_entity(entity2)
+
+                f = "Value lt @value"
+                params = {"value": 25}
```

Comment:
What happens if I query with a 6-decimal-place datetime value?

Reply:
The entity properties are created as strings, the query will have to treat the entity property as a string and do an `eq` or similar string-friendly query.

Reply:
Nothing to do here - as we can't format the datetime query for them - they will need to do it themselves.